src/category_theory/limits/presheaf.lean

```diff
@@ -0,0 +1,182 @@
+/-
+Copyright (c) 2020 Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bhavik Mehta
+-/
+import category_theory.adjunction
+import category_theory.elements
+import category_theory.limits.functor_category
+import category_theory.limits.shapes
+import category_theory.limits.types
+
+namespace category_theory
+
+noncomputable theory
+
+open category limits
+universes u₁ u₂
+
+variables {C : Type u₁} [small_category C]
+variables {ℰ : Type u₂} [category.{u₁} ℰ]
+variable (A : C ⥤ ℰ)
+
+namespace colimit_adj
+
+/--
+The functor taking `(E : ℰ) (c : Cᵒᵖ)` to the homset `(A.obj C ⟶ E)`. It is shown in `L_adjunction`
+that this functor has a left adjoint (provided `E` has colimits) given by taking colimits over
+categories of elements.
+-/
+@[simps {rhs_md := semireducible}]
+def R : ℰ ⥤ (Cᵒᵖ ⥤ Type u₁) :=
```

Comment:
Should this be called something with `yoneda` in the name? @rwbarton Do you have a good name suggestion for this functor?
`restricted_yoneda`?

Reply:
I used this, I'm happy to change if anyone has a strong preference for something else.